firebase/deployment/src/main/java/io/quarkiverse/googlecloudservices/firebase/deployment/FirebaseDevServiceConfig.java

```diff
@@ -0,0 +1,282 @@
+package io.quarkiverse.googlecloudservices.firebase.deployment;
+
+import java.util.Optional;
+
+import io.quarkus.runtime.annotations.ConfigRoot;
+import io.smallrye.config.ConfigMapping;
+import io.smallrye.config.WithDefault;
+
+/**
+ * Root configuration class for Google Cloud Firebase that operates at build time.
+ * This class provides a nested structure for configuration, including
+ * a separate group for the development service configuration.
+ * <p>
+ * Here is an example of how to configure these properties:
+ * <p>
+ *
+ * <pre>
+ * quarkus.google.cloud.firebase.devservice.enabled = true
+ * quarkus.google.cloud.pubsub.devservice.image-name = gcr.io/google.com/cloudsdktool/google-cloud-cli # optional
+ * quarkus.google.cloud.pubsub.devservice.emulatorPort = 8085 # optional
+ * </pre>
+ */
+@ConfigMapping(prefix = "quarkus.google.cloud")
```

Comment:
```suggestion
@ConfigMapping(prefix = "quarkus.google.cloud.firebase-dev-services")
```

Reply:
You can read a config from another extension in an extension.

But we may need an SPI here if we need to access something from this extension from other extensions.
An SPI module is module that expose an autonomous module (with no dependencies) to other extensions for ex to access shared configuration or information. Usually in the form of a build item.

As this PR is already big, we can do this later, but the config root must change

Reply:
Ok clear, can work on that as a next PR. Working on a follow up anyway, so should be able to do that fairly quickly. Can we keep the config root for this PR? If we change it, the setup won't work and it'll result in conflicts with the Firestore and Pubsub dev services.

Reply:
@loicmathieu any update? In the mean time I'm finishing up work on a next PR to be able to include the firebase.json (and other files used by a regular emulator) inside the docker container (easing development). Did some refactoring too which seperates out the TestContainers stuff from the Quarkus dev service/integration stuff. (See https://github.com/jfbenckhuijsen/quarkus-google-cloud-services/pull/1)

Next up will be a PR to get that SPI Module running. Just for my understanding, that's just an extension module (set of -deployment and -runtime) without dependencies to other modules which is used by others?

I suggest I'll create the SPI stuff as a PR onto this PR, so we can split up reviewing into separate PRs just like the one I mentioned in this comment. Once approved, I'll merge them into this PR, and we can merge this one into the main branch?

Reply:
> Can we keep the config root for this PR?

No, it really need to use a separate root config, but as I said, you can access any config option from any dependency so this should not be an issue.

Reply:
Created #720